framework/include/I2CDevObj.hpp

```diff
@@ -70,6 +70,9 @@ class I2CDevObj : public DevObj
 	static int readReg(DevHandle &h, uint8_t address, uint8_t *out_buffer, int length);
 	static int writeReg(DevHandle &h, uint8_t address, uint8_t *in_buffer, int length);
 
+	// read from a register without ioctl
+	int simple_read(uint8_t *out_buffer, int length);
```

Comment:
`simple_read` should be protected and called `_simple_read` The public ones need to support the interface with the DevHandle.